api-docs/swagger.yml

```diff
@@ -263,53 +276,160 @@ paths:
           description: Healthy
         '500':
           description: Not healthy
+securityDefinitions:
+  basicAuth:
+    type: basic
 definitions:
+  Info:
+    type: object
+    properties:
+      provider:
+        type: string
+        description: The OpenFaaS Provider
+        example: faas-swarm
+      version:
+        type: object
+        description: Version of the OpenFaaS Provider
+        properties:
+          sha:
+            type: string
+            example: 7108418d9dd6b329ddff40e7393b3166f8160a88
+          release:
+            type: string
+            format: semver
+            example: 0.2.6
+      orchestration:
+        type: string
+        example: swarm
+    required:
+      - provider
+      - version
+      - orchestration
   DeleteFunctionRequest:
     type: object
     properties:
       functionName:
         type: string
         description: Name of deployed function
-  CreateFunctionRequest:
+        example: nodeinfo
+    required:
+      - functionName
+  FunctionDefintion:
     type: object
     properties:
       service:
         type: string
         description: Name of deployed function
+        example: nodeinfo
       network:
         type: string
-        description: 'Docker swarm network, usually func_functions'
+        description: Docker swarm network, usually func_functions
+        example: func_functions
       image:
         type: string
         description: Docker image in accessible registry
+        example: functions/nodeinfo:latest
       envProcess:
         type: string
         description: Process for watchdog to fork
+        example: node main.js
       envVars:
         type: object
         additionalProperties:
           type: string
         description: Overrides to environmental variables
-      labels:
+      constraints:
         type: array
         items:
           type: string
+          description: Constraints are specific to OpenFaaS Provider
+          example: "node.platform.os == linux"
+      labels:
+          type: string
+        items:
+        type: array
         description: An array of labels used by the back-end for making scheduling or routing decisions
       annotations:
-        type: array
-        items:
           type: string
         description: An array of annotations used by the back-end for management, orchestration, events and build tasks
+        items:
+        type: array
       secrets:
         type: array
         items:
           type: string
-        description: An array of names of secrets that are required to be loaded from the Docker Swarm.
+          description: An array of names of secrets that are required to be loaded from the Docker Swarm.
+          example: "secret-name-1"
       registryAuth:
         type: string
         description: >-
           Private registry base64-encoded basic auth (as present in
           ~/.docker/config.json)
+        example: Basic dXNlcjpwYXNzd29yZA==
```

Comment:
I think this example is incorrect, it should not have a prefix of "Basic "

Reply:
Oh I see, adjusted it in 5c657e6